draft-ietf-httpbis-digest-headers.md

```diff
@@ -635,6 +635,13 @@ or by sending multiple representation-data-digest values from which the receiver
 Endpoints are advised that sending multiple values consumes resources,
 which may be wasted if the receiver ignores them (see {{digest}}).
```

Comment:
The point I believe @jricher made in #1841 is that it's just as easy to downgrade to weak algorithms by stripping off better ones. Since digest has no in built mechanisms to prevent downgrade, if people care then they should protect the digest fields. That could be done, for example, by using TLS or digital signatures. It could also be done by endpoints restricting the algorithms that they support - potentially rejecting processing of messages that do not contain a digest contained in their supported set,

The algorithm substitution attack mentioned in the next paragraph and described in RFC6211 seems to be something a bit different. It seems to predicate on the verification depending on the length of the hash, not the so much algorithm itself. Since the algorithm identifier is always included in digest, I don't know how much this one applies. Perhaps if there is a digest algorithm that encodes its parameters in the representation-data-digest, as we suggest in Section 1.3. But again the only way to protect from manipulation of field values is to use something like TLS or sigs.

So how about rephrasing the paragraph addition to something like:

While algorithm agility allows the migration to stronger algorithms it does not prevent the use of weaker algorithms. Digest and Content-Digest do not provide any mitigiations for downgrade or substitution (see Section 1 of {{?RFC6211}} attacks of the digest-algorithm. To protect against such attacks, endpoints could restrict their set of supported algorithms to stronger ones and protect the value of Digest and Content-Digest by using TLS and/or digital signatures.

Reply:
@LPardue @jricher PTAL